From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Mon, 19 Feb 2024 22:25:23 +0000
Subject: [PATCH] Add support for calling added APIs from custom OS

Adapted from GrapheneOS Vanadium patch 0158-Add-support-for-calling-added-APIs-from-custom-OS.patch for the Vandium browser port.

---
 vandium/ext/custom_os/BUILD.gn               | 13 +++++++++
 .../src/app/vandium/ext/CustomOSApis.java    | 29 +++++++++++++++++++
 2 files changed, 42 insertions(+)
 create mode 100644 vandium/ext/custom_os/BUILD.gn
 create mode 100644 vandium/ext/custom_os/java/src/app/vandium/ext/CustomOSApis.java

diff --git a/vandium/ext/custom_os/BUILD.gn b/vandium/ext/custom_os/BUILD.gn
new file mode 100644
index 0000000000000..ea3c3995a44e5
--- /dev/null
+++ b/vandium/ext/custom_os/BUILD.gn
@@ -0,0 +1,13 @@
+# Copyright 2023 GrapheneOS
+# Use of this source code is governed by a GPL-2.0 style license that can be
+# found in the LICENSE file.
+
+import("//build/config/android/rules.gni")
+
+android_library("custom_os_apis_java") {
+  sources = [
+    "java/src/app/vandium/ext/CustomOSApis.java",
+  ]
+  deps = [
+  ]
+}
\ No newline at end of file
